src/components/SizeObserver/SizeObserver.tsx

```diff
@@ -0,0 +1,120 @@
+/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
+import '../../yamui';
+import * as React from 'react';
+import { BaseComponentProps } from '../../util/BaseComponent/props';
+
+export interface SizeObserverProps extends BaseComponentProps {
+  /**
+   * An object of `size` keys mapped to height/width queries. Example:
+   *
+   *     {
+   *        small: {
+   *          maxWidth: 500,
+   *        },
+   *        large: {
+   *          minWidth: 501,
+   *        },
+   *     }
+   */
+  query: SizeQueriesMap;
+
+  /**
+   * A render prop which receives the key of the active query.
+   */
+  renderAt(size?: string): React.ReactNode;
+}
+
+export interface SizeObserverState {
+  currentSize?: string;
+}
+
+export interface SizeQueries {
+  maxWidth?: number;
+  minWidth?: number;
+  maxHeight?: number;
+  minHeight?: number;
+}
+
+export interface SizeQueriesMap {
+  [key: string]: SizeQueries;
+}
+
+const getMatchingQuerySize = (query: SizeQueriesMap, height: number, width: number) => {
+  for (const key of Object.keys(query)) {
+    const { maxHeight, minHeight, maxWidth, minWidth } = query[key];
+
+    if (maxHeight !== undefined && height > maxHeight) {
+      continue;
+    }
+    if (minHeight !== undefined && height < minHeight) {
+      continue;
+    }
+    if (maxWidth !== undefined && width > maxWidth) {
+      continue;
+    }
+    if (minWidth !== undefined && width < minWidth) {
+      continue;
+    }
+
+    return key;
+  }
```

Comment:
It took me a little while to grok this with the `continue`s, how would you feel about this refactor?
```
const getMatchingQuerySize = (query: SizeQueriesMap, height: number, width: number) => {
  for (const key of Object.keys(query)) {
    const {
      maxHeight = Number.MAX_SAFE_INTEGER,
      maxWidth = Number.MAX_SAFE_INTEGER,
      minHeight = 0,
      minWidth = 0,
    } = query[key];
    if (height > maxHeight && height < minHeight && width > maxWidth && width < minWidth) {
      return key;
    }
  }
};
```

Reply:
I agree this is more understandable, thanks! We will update.